src/utils/formatting.js

```diff
@@ -0,0 +1,14 @@
+// takes fileSize in KBs with 3 decimals places, and formats them
+// if size < 100, display size in KBs to 2dps, e.g. 19.432 --> '19.43KB'
+// if size <= 999.499, display size in KBs rounded to nearest integer, e.g. 431.855 --> '432KB'
+// if size > 999.499, display size in MBs to 2ps, e.g. 12521.482 --> 12.52MB'
```

Comment:
I would put these precise comments in the function to explain each `if` block, and have a more global description for the function here.

Reply:
good call!